src/LIBSVM.jl

```diff
@@ -323,14 +378,15 @@ function svmtrain(
     ninstances = size(X, 2)
 
     # Construct SVMProblem
-    if kernel == Kernel.Precomputed
-        (nodes, nodeptrs) = gram2nodes(X)
-
-        # This is necessary to construct SupportVectors correctly
-        # The struct needs to contain the indices of support vectors
-        X = (1:size(X, 1))'
```

Comment:
Is this not required anymore?

Reply:
Oh, I see.

The reason is the way libsvm handles precomputed kernels. See [this line](https://github.com/cjlin1/libsvm/blob/master/svm.cpp#L249) which simply assumes that `x[j][0]` contains the index of `j`-th instance. I am not sure why they decided to do it like that.

Reply:
I added your explanation as a comment in the code.